manim/mobject/svg/text_mobject.py

```diff
@@ -1064,3 +1065,205 @@ def text2svg(self):
+            each.clear_points()
+            for index, point in enumerate(points):
+                each.append_points([point])
+                if (
+                    index != len(points) - 1
+                    and (index + 1) % nppc == 0
+                    and any(point != points[index + 1])
+                ):
+                    each.add_line_to(last)
+                    last = points[index + 1]
+            each.add_line_to(last)
+
+        if self.gradient:
+            self.set_color_by_gradient(*self.gradient)
+        for col in colormap:
+            self.chars[col["start"] : col["end"]].set_color(
+                Colors[col["color"].lower()].value
+            )
+        for grad in gradientmap:
+            self.chars[grad["start"] : grad["end"]].set_color_by_gradient(
+                *(Colors[grad["from"].lower()].value, Colors[grad["to"].lower()].value)
+            )
+        # anti-aliasing
+        if self.height is None and self.width is None:
+            self.scale(TEXT_MOB_SCALE_FACTOR)
+
+    def text2hash(self):
+        """Internally used function.
+        Generates ``sha256`` hash for file name.
+        """
+        settings = (
+            "MARKUPPANGO" + self.font + self.slant + self.weight
+        )  # to differentiate from classical Pango Text
+        settings += str(self.line_spacing) + str(self.size)
+        id_str = self.text + settings
+        hasher = hashlib.sha256()
+        hasher.update(id_str.encode())
+        return hasher.hexdigest()[:16]
+
+    def text2svg(self):
+        """Internally used function.
+        Convert the text to SVG using Pango
+        """
+        size = self.size * 10
+        line_spacing = self.line_spacing * 10
+        dir_name = config.get_dir("text_dir")
+        disable_liga = self.disable_ligatures
+        if not os.path.exists(dir_name):
+            os.makedirs(dir_name)
+        hash_name = self.text2hash()
+        file_name = os.path.join(dir_name, hash_name) + ".svg"
+        if os.path.exists(file_name):
+            return file_name
+        surface = cairocffi.SVGSurface(file_name, 600, 400)
+        context = cairocffi.Context(surface)
+        context.move_to(START_X, START_Y)
+        offset_x = 0
+        last_line_num = 0
+        layout = pangocairocffi.create_layout(context)
+        layout.set_width(pangocffi.units_from_double(600))
+
+        fontdesc = pangocffi.FontDescription()
+        fontdesc.set_size(pangocffi.units_from_double(size))
+        if self.font:
+            fontdesc.set_family(self.font)
+        fontdesc.set_style(self.str2style(self.slant))
+        fontdesc.set_weight(self.str2weight(self.weight))
+        layout.set_font_description(fontdesc)
+
+        text = self.text.replace("\n", " ")
+        context.move_to(START_X, START_Y)
+        pangocairocffi.update_layout(context, layout)
+        if disable_liga:
+            layout.set_markup(f"<span font_features='liga=0'>{text}</span>")
```

Comment:
A suggestion here. 

Currently, the passing of character which shouldn't be passed to an XML file will create an error here. for example `<`,`>`.
We can use the function escape from `xml.sax.saxutils` to file it. See what it done here
https://github.com/ManimCommunity/manim/blob/ec7a962d620ebd89a8a2d5ac89d8c164b893262f/manim/mobject/svg/text_mobject.py#L1057-L1059

Reply:
You are right, but I cannot use `escape`, because the string contains (or may contain) special characters for formatting. I think this will be a matter of documentation: we have to make clear that `&` and `<` have to be written as `&amp;` and `&lt;`. For `>` it works, as shown in one example.